File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* removeNthFromEnd(ListNode* head, int n) {
        ListNode* dummy = new ListNode();
        dummy->next = head;
        
        ListNode* s, *f;
        s = f = dummy;
        
        for(int i=1; i<=n; i++)
        {
            f = f->next;
        }
        
        while(f->next != NULL)
        {
            f = f->next;
            s = s->next;
        }
        
        s->next = s->next->next;
        
        return dummy->next;
    }
};

File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
        slow = fast = head
        itr = 0
        while itr != n:
            fast = fast.next
            itr = itr + 1
        
        if fast is None:
            return head.next
        
        while fast.next != None:
            slow = slow.next
            fast = fast.next
        
        slow.next = slow.next.next
        return head
        